### PR TITLE
test(footer): migrate to playwright

### PR DESCRIPTION
**Pull request by brandyscarney**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [ ] Build (`npm run build`) was run locally and any changes were pushed
- [ ] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Migration to playwright

### CODE REVIEW

**Comment by brandyscarney** [comment]
@amandaejohnston I did yesterday but it hates me for some reason 😩 https://github.com/ionic-team/ionic-framework/actions/workflows/update-screenshots.yml?query=branch%3AFW-1205

**Comment by amandaejohnston** [comment]
Looks like one of the tests is failing; it's probably just flaky since only one of the shards failed. (If there were an actual test failure, you'll usually see at least 3-4 borked shards.) If you click on the test run, you can see more details on what went wrong. I would try just re-running the failed jobs using the dropdown at the top right.

File: core/src/components/footer/test/fade/footer.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  test('should not have visual regressions with fade footer', async ({ page, skip }) => {
+    skip.mode('md', 'Translucent effect is only available in iOS mode.');
+    skip.browser('firefox', 'Firefox has some issues rendering translucent effects on Linux.');
+
+    await page.goto('/src/components/footer/test/fade');
+
+    const footer = page.locator('ion-footer');
+    expect(await footer.screenshot()).toMatchSnapshot(`footer-fade-blurred-diff-${page.getSnapshotSettings()}.png`);
+
+    const content = (await page.$('ion-content'))!;
```
Comment (amandaejohnston) [inline core/src/components/footer/test/fade/footer.e2e.ts:18]:
```suggestion
    const content = page.locator('ion-content');
```
See: https://playwright.dev/docs/locators#locator-vs-elementhandle

File: core/src/components/footer/test/fade/footer.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+          padding: 50px 0;
+        }
+      </style>
+    `);
+
+    const footer = page.locator('ion-footer');
+    expect(await footer.screenshot()).toMatchSnapshot(`footer-fade-scroll-target-blurred-diff-${page.getSnapshotSettings()}.png`);
+
+    const scrollTarget = (await page.$('#scroll-target'))!;
```
Comment (amandaejohnston) [inline core/src/components/footer/test/fade/footer.e2e.ts:81]:
```suggestion
    const scrollTarget = page.locator('#scroll-target');
```

**Review by amandaejohnston — APPROVED** [review]
LGTM once the comments are addressed 👍

**Pull request merged**